File: 프로그래머스/LV2/코딩테스트 연습/SummerWinter Coding(~2018)/영어 끝말잇기/영어 끝말잇기.cpp
#include <string>
#include <vector>
#include <set>

using namespace std;

vector<int> solution(int n, vector<string> words) 
{
    vector<int> answer = {0,0};
    set<string> used;
    string before = words[0];
    used.insert(words[0]);

    for(int i = 0; i < words.size(); i++)
    {
        if(before.back() != words[i][0] || used.find(words[i]) != used.end())
        {
            answer[0] = i%n+1;
            answer[1] = i/n+1;
            break;
        }
        used.insert(words[i]);
        before = words[i];
    }
    return answer;
}

File: 프로그래머스/LV2/코딩테스트 연습/SummerWinter Coding(~2018)/영어 끝말잇기/영어 끝말잇기.py
def solution(n, words):
    answer = [0,0]
    used = set()
    before = words[0]
    used.add(words[0])

    for i in range(1, len(words)):
        if before[-1] != words[i][0] or words[i] in used:
            answer = [i%n+1, i//n+1]
            break
        used.add(words[i])
        before = words[i]
    return answer